web/react/components/channel_notifications_modal.jsx

```diff
@@ -32,11 +32,13 @@ export default class ChannelNotificationsModal extends React.Component {
             activeSection: ''
         };
     }
-    componentDidMount() {
-        ChannelStore.addChangeListener(this.onListenerChange);
-    }
-    componentWillUnmount() {
-        ChannelStore.removeChangeListener(this.onListenerChange);
+    componentWillReceiveProps(nextProps) {
+        if (nextProps.show) {
```

Comment:
This should be `nextProps.show && !this.props.show` or else we might get this bound multiple times as a listener